# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/BytecodeControlStructures.cpp
# Commit: 19f956b37c6c
# Full Hash: 19f956b37c6cb832b41b8d1ffb8648f38336f4dd
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-19 21:52:02
# Description:
#   Bug 1602392 - Change for-of bytecode to not have an extra value on the stack across the loop edge. r=arai
#   
#   This fixes a bug with HasLiveStackValueAtDepth because now all stack values that
#   are part of the loop (2 values instead of 3 values) are loop-invariant.
#   
# ==============================================================================

diff -r f1547816cf32 -r 19f956b37c6c js/src/frontend/BytecodeControlStructures.cpp
--- a/js/src/frontend/BytecodeControlStructures.cpp	Thu Dec 19 10:57:13 2019 +0000
+++ b/js/src/frontend/BytecodeControlStructures.cpp	Thu Dec 19 09:06:01 2019 +0000
@@ -52,19 +52,6 @@
   return bce->emitJumpTargetAndPatch(continues);
 }
 
-bool LoopControl::emitSpecialBreakForDone(BytecodeEmitter* bce) {
-  // This doesn't pop stack values, nor handle any other controls.
-  // Should be called on the toplevel of the loop.
-  MOZ_ASSERT(bce->bytecodeSection().stackDepth() == stackDepth_);
-  MOZ_ASSERT(bce->innermostNestableControl == this);
-
-  if (!bce->emitJump(JSOP_GOTO, &breaks)) {
-    return false;
-  }
-
-  return true;
-}
-
 bool LoopControl::emitLoopHead(BytecodeEmitter* bce,
                                const Maybe<uint32_t>& nextPos) {
   // Insert a NOP if needed to ensure the script does not start with a